src/main/jbake/content/blog/2017/07/jaeger-with-security-proxy.adoc

```diff
@@ -0,0 +1,253 @@
+
+We'll use the link:https://github.com/jaegertracing/jaeger-openshift/blob/master/production/jaeger-production-template.yml[Jaeger OpenShift Production template]
+as the starting point: either clone the entire repository, or just get a local version of the template.
+
+The first step is to add the sidecar container to the `query-deployment` object. Under the `containers` list, after we
+specify the `jaeger-query`, let's add the sidecar:
+
+```yml
+        - image: jpkroehling/keycloak-proxy
+          name: ${JAEGER_SERVICE_NAME}-query-security-proxy
+          volumeMounts:
+          - mountPath: /opt/jboss/conf
+            name: security-proxy-configuration-volume
+          ports:
+          - containerPort: 8080
+            protocol: TCP
+          readinessProbe:
+            httpGet:
+              path: "/"
+              port: 8080
+```
+
+Note that container specifies a `volumeMount` named `security-proxy-configuration-volume`: we'll use it to store the
+proxy's configuration file. The volume should be added under the `spec` node for the deployment's template and is a sibling
+node to the `dnsPolicy` property:
+
+```yml
+        volumes:
+          - configMap:
+              name: ${JAEGER_SERVICE_NAME}-configuration
+              items:
+                - key: proxy
+                  path: proxy.json
+            name: security-proxy-configuration-volume
+```
+
+Now, we need to specify the `ConfigMap`, with the proxy's configuration entry. To do that, we add a new top-level item to
+the template:
+
+```yml
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: ${JAEGER_SERVICE_NAME}-configuration
+    labels:
+      app: jaeger
+      jaeger-infra: security-proxy-configuration
+  data:
+    proxy: |
+      {
+          "target-url": "http://localhost:16686",
+          "bind-address": "0.0.0.0",
+          "http-port": "8080",
+          "applications": [
+              {
+                  "base-path": "/",
+                  "adapter-config": {
+                    "realm": "jaeger",
+                    "auth-server-url": "${REDHAT_SSO_URL}",
+                    "ssl-required": "external",
+                    "resource": "proxy-jaeger",
+                    "credentials": {
+                      "secret": "THE-SECRET-FROM-INSTALLATION-FILE"
+                    }
+                  }
+            ,
+            "constraints": [
+                      {
+                          "pattern": "/*",
+                          "roles-allowed": [
+                              "user"
+                          ]
+                      }
+                  ]
+              }
+          ]
+}
+```
+
+Note that we are only allowing users with the role `user` to log into our Jaeger UI. In a real world scenario, you might
+want to adjust this to fit your setup. For instance, your user data might come from LDAP and you'd want to allow all users
+from specific LDAP groups to access Jaeger UI.
```

Comment:
For instance, your user data might come from LDAP and you'd want to allow all users from specific LDAP groups to access Jaeger UI.

Suggestion - For instance, your user data might come from LDAP, and you only want to allow users from specific LDAP groups to access the Jaeger UI.